### PR TITLE
SALTO-1192: Fixed validation errors in workbook and translationcollection

### PR DESCRIPTION
**Pull request by alonstern**
Fixed validation error on the name value in translationcollection, workbook, and dataset types

---
_Release Notes_: 
Bugfixes:
Netsuite Adapter:
Fixed validation error of values that contain '#text' in them

---
_User Notifications_: 
new "translate fields" will be added to some of the types

Values of the form 
```
name = {
    '#text' = 'value'
    translate = 'T'
}
```
will be changed to
```
name = 'value'
nameTranslate = true
```
(This change will be omitted from the tracker)

### CODE REVIEW

File: packages/netsuite-adapter/src/filters/name_translation_converter.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          change,
+          async instance => {
+            if (TYPES_TO_CONVERT.every(type => type.elemID.name !== instance.elemID.typeName)
+              || instance.value.nameTranslate === undefined) {
+              return instance
+            }
+            instance.value.name = {
+              '#text': instance.value.name,
+              translate: instance.value.nameTranslate ? 'T' : 'F',
+            }
```
Comment (omrilit) [inline packages/netsuite-adapter/src/filters/name_translation_converter.ts:62]: Did you test that the deploy of these instances work when having this format?

**Review by omrilit — COMMENTED** [review]
When thinking about it, it means that upon user's next fetch, in addition to the type changes of the above 3 types, there will be changes in their instances as well.
We should consider to ignore them in the next tracker's notification.
We should also consider to update customers that it's about to appear in their next NS deploy in case these types are not in skipList / they have such instances in their account.

**Comment by omrilit** [comment]
Please see my comment in the related ticket, I think we should consider a different approach here as it seems to be a more general case cross many/all records.
https://salto-io.atlassian.net/browse/SALTO-1192?focusedCommentId=20719

**Comment by alonstern** [comment]
@omrilit This PR is ready for re-review (still needs to update the tracker in a different PR in the SaaS)

**Review by omrilit — APPROVED** [review]
LGTM, only few small nits

File: packages/netsuite-adapter/src/filters/translation_converter.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          values: instance.value,
+          type: await instance.getType(),
+          strict: false,
+          pathID: instance.elemID,
+          transformFunc: ({ value }) => {
+            if (!_.isPlainObject(value)) {
+              return value
+            }
+            _(value)
+              .entries()
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/netsuite-adapter/src/filters/translation_converter.ts:87]: For some reason when I do it the filter in line 88 thinks `val` is `unknown` and not `Value`

**Pull request merged**